server/src/test/java/io/crate/planner/optimizer/rule/MoveFilterBeneathJoinTest.java

```diff
@@ -0,0 +1,191 @@
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.optimizer.rule;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import io.crate.analyze.WhereClause;
+import io.crate.analyze.relations.AbstractTableRelation;
+import io.crate.analyze.relations.AnalyzedRelation;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.RelationName;
+import io.crate.planner.operators.Collect;
+import io.crate.planner.operators.Filter;
+import io.crate.planner.operators.HashJoin;
+import io.crate.planner.optimizer.costs.PlanStats;
+import io.crate.planner.optimizer.matcher.Captures;
+import io.crate.planner.optimizer.matcher.Match;
+import io.crate.statistics.TableStats;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SqlExpressions;
+import io.crate.testing.T3;
+
+import static io.crate.testing.Asserts.assertThat;
+
+public class MoveFilterBeneathJoinTest extends CrateDummyClusterServiceUnitTest {
+
+    private SqlExpressions sqlExpressions;
+    private Map<RelationName, AnalyzedRelation> sources;
+    private PlanStats planStats;
+    private Collect c1;
+    private Collect c2;
+    private Collect c3;
+
+    @Before
+    public void prepare() throws Exception {
+        sources = T3.sources(clusterService);
+        sqlExpressions = new SqlExpressions(sources);
+        planStats = new PlanStats(sqlExpressions.nodeCtx,
+                                  CoordinatorTxnCtx.systemTransactionContext(),
+                                  new TableStats());
+
+        var t1 = (AbstractTableRelation<?>) sources.get(T3.T1);
+        var t2 = (AbstractTableRelation<?>) sources.get(T3.T2);
+        var t3 = (AbstractTableRelation<?>) sources.get(T3.T3);
+
+        c1 = new Collect(t1, Collections.emptyList(), WhereClause.MATCH_ALL);
+        c2 = new Collect(t2, Collections.emptyList(), WhereClause.MATCH_ALL);
+        c3 = new Collect(t3, Collections.emptyList(), WhereClause.MATCH_ALL);
```

Comment:
Can't we use the `SQLExecuter.builder` instead? This would be much more readable...
```suggestion

        SQLExecutor.builder(clusterService)
            .addTable("create table t1 (a int)")
            .addTable("create table t2 (b int)")
            .addTable("create table t3 (c int)")
            .build();
        c1 =  e.logicalPlan("SELECT a FROM t1");
        c2 = e.logicalPlan("SELECT b FROM t2");
        c3 = e.logicalPlan("SELECT c FROM t3");
```